### PR TITLE
feat: Refactor to support templating

### PR DESCRIPTION
**Pull request by ghost**
This PR introduces a rename of https://github.com/theopensystemslab/planx-document-review (i.e. it is a clone of that repo). It also restructures this library to take ownership of streaming.

The React documents have been updated to be a little more robust (i.e. a proper html structure for a standalone page). There's room here for multiple types of templating tools to be used (i.e. Docx) but it still to be confirmed if these formats will be useful.

### CODE REVIEW

**Review by DafyddLlyr — APPROVED** [review]
This is solid code - I really like your style here 👌 The tests were helpful and descriptive which was great also.

I support bringing `renderToPipeableStream()` in house - that feels right from a responsibility point of view to me.

I've not taken a close look at the template itself yet, but am happy to do so when you let me know that it's ready.

File: src/index.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+export function generateDocxTemplateStream({
+  templateName,
+  passport,
+}: {
+  templateName: string;
+  passport: Passport;
+}) {
+  if (!hasRequiredDataForTemplate({ templateName, passport })) {
+    throw new Error(`Template "${templateName}" is missing required fields`);
```
[the remark this answers is not in the document]
Comment (DafyddLlyr) [reply src/index.tsx:50]: Yeah I saw that was exposed - that's a good design decision.

File: src/overview/helpers.ts [hunk 2]
```diff
@@ -17,4 +18,4 @@ export function safeDecodeURI(data: string): string {
   } catch (error) {
     return data
   };
-};
\ No newline at end of file
+};
```
Comment (jessicamcinchak) [inline src/overview/helpers.ts:21]: i'm admittedly no typescript expert - but I think this type of helper function to essentially further-check a data structure that we've also defined a custom type/interface for is maybe a good fit for a user-defined type guard like described here? https://basilebong.com/blog/custom-type-guards-in-typescript-dg8

had to read this one over a couple times to understand it & took me a bit to wrap my head around why the `PlanXExportData[]` type isn't valid "enough" on its own - maybe the type guard would help with clarity? just a thought, might not be the right approach!

**Review by jessicamcinchak — APPROVED** [review]
This seems like a really thoughtful refactor! Really nice to see the first go at the doc mappings & file generation too. 

Thanks especially for so many test and examples - are those actually running on CI yet here? We have a pnpm/vitest [github workflow file](https://github.com/theopensystemslab/map/blob/main/.github/workflows/pnpm-test.yml) in the map repo that might be quick to copy over for these too if you want, just fyi!

**Pull request merged**